clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,317 @@
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/execservice"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/checker"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/runtimeobjects"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/workerqueue"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	serviceCreationLatencyName           = "ServiceCreationLatency"
+	serviceCreationLatencyWorkers        = 10
+	defaultServiceCreationLatencyTimeout = 1 * time.Minute
+	defaultWaitTimeout                   = 10 * time.Minute
+	defaultCheckInterval                 = 10 * time.Second
+	pingBackoff                          = 1 * time.Second
+	pingChecks                           = 10
+
+	creatingPhase     = "creating"
+	ipAssigningPhase  = "ipAssigning"
+	reachabilityPhase = "reachability"
+)
+
+func init() {
+	measurement.Register(serviceCreationLatencyName, createServiceCreationLatencyMeasurement)
+}
+
+func createServiceCreationLatencyMeasurement() measurement.Measurement {
+	return &serviceCreationLatencyMeasurement{
+		selector:      measurementutil.NewObjectSelector(),
+		queue:         workerqueue.NewWorkerQueue(serviceCreationLatencyWorkers),
+		creationTimes: measurementutil.NewObjectTransitionTimes(serviceCreationLatencyName),
+		pingCheckers:  checker.NewCheckerMap(),
+	}
+}
+
+type serviceCreationLatencyMeasurement struct {
+	selector      *measurementutil.ObjectSelector
+	waitTimeout   time.Duration
+	stopCh        chan struct{}
+	isRunning     bool
+	queue         workerqueue.Interface
+	client        clientset.Interface
+	creationTimes *measurementutil.ObjectTransitionTimes
+	pingCheckers  checker.CheckerMap
+}
+
+// Execute executes service startup latency measurement actions.
+// Services can be specified by field and/or label selectors.
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+// "start" action starts observation of the services.
+// "waitForReady" waits until all services are reachable.
+// "gather" returns service created latency summary.
+func (s *serviceCreationLatencyMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	s.client = config.ClusterFramework.GetClientSets().GetClient()
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if s.selector.Parse(config.Params); err != nil {
+			return nil, err
+		}
+		s.waitTimeout, err = util.GetDurationOrDefault(config.Params, "waitTimeout", defaultServiceCreationLatencyTimeout)
+		if err != nil {
+			return nil, err
+		}
+		return nil, s.start()
+	case "waitForReady":
+		err = s.waitForReady()
+		if err != nil {
+			klog.Errorf("%s: %v", s, err)
+		}
+		return nil, err
```

Comment:
don't log error if you return it - it's responsibility of a caller to handle it.
This whole case should be:
```
return nil, s.waitForReady()
```

Reply:
We have similar logic on all other measurements. If I'm changing this, I have to change it in all other places.
I'll keeping it this way (to keep style coherent) and I'll add the issue to fix the problem on cluster loader level.

Reply:
#682

Reply:
Given you're inconsistent already anyway, let's change this one now.

Reply:
Done.